crates/neon-build/src/napi/mod.rs

```diff
@@ -0,0 +1,173 @@
+use std::path::{Path, PathBuf};
+
+fn manifest_dir() -> PathBuf {
+    std::env::var_os("CARGO_MANIFEST_DIR")
+        .map(PathBuf::from)
+        .expect("Expected CARGO_MANIFEST_DIR environment variable")
+}
+
+fn out_dir() -> PathBuf {
+    std::env::var_os("OUT_DIR")
+        .map(PathBuf::from)
+        .expect("Expected OUT_DIR environment variable")
+}
+
+fn is_env(env_var: &str, value: &str) -> bool {
+    std::env::var_os(env_var)
+        .map(|v| v == value)
+        .unwrap_or(false)
+}
+
+fn setup_unix(output_file: PathBuf) {
+    println!("cargo:rustc-cdylib-link-arg=-o");
+    println!("cargo:rustc-cdylib-link-arg={}", output_file.display());
+}
+
+fn setup_windows(output_file: PathBuf) {
+    let pdb_file = output_file
+        .file_name()
+        .map(|file| out_dir().join(Path::new(file).with_extension("pdb")))
+        .expect("Expected a neon output file name");
+
+    println!("cargo:rustc-cdylib-link-arg=/OUT:{}", output_file.display());
+    println!("cargo:rustc-cdylib-link-arg=/PDB:{}", pdb_file.display());
+}
+
+/// `Builder` acts as a builder for initializing a Neon build script
+///
+/// A default setup builder is provided as [`neon_build::setup()`](crate::setup()).
+///
+/// # Example
+///
+/// Output the neon module at `lib/native.node`
+///
+/// ```
+/// # #[allow(clippy::needless_doctest_main)]
+/// fn main() {
+///     neon_build::Builder::new()
+///         .output_dir("lib")
+///         .output_dir("native.node")
+///         .setup();
+/// }
+#[derive(Debug, Default)]
+pub struct Builder {
+    output_dir: Option<PathBuf>,
+    output_file: Option<PathBuf>,
+}
+
+impl Builder {
```

Comment:
I think `Builder` is a confusing name since it doesn't describe what it does -- and it also could be confusing because this crate is `neon_build` but `Builder` isn't about running a Neon build.

We talked about `Setup::options()` being the way to construct this thing. I think it reads nicely.